prql-compiler/tests/integration/main.rs

```diff
@@ -74,21 +89,124 @@ impl IntegrationTest for Dialect {
                 config.trust_cert();
                 config.authentication(AuthMethod::sql_server("sa", "Wordpass123##"));
 
-                Box::new(
-                    RUNTIME
-                        .block_on(async {
-                            let tcp = TcpStream::connect(config.get_addr()).await?;
-                            tcp.set_nodelay(true).unwrap();
-                            Client::connect(config, tcp.compat_write()).await
-                        })
-                        .unwrap(),
-                )
-            }),
+                Some(DbConnection {
+                    dialect: Dialect::MsSql,
+                    protocol: Box::new(
+                        RUNTIME
+                            .block_on(async {
+                                let tcp = TcpStream::connect(config.get_addr()).await?;
+                                tcp.set_nodelay(true).unwrap();
+                                Client::connect(config, tcp.compat_write()).await
+                            })
+                            .unwrap(),
+                    ),
+                })
+            }
             _ => None,
         }
     }
 }
 
+impl SetUpData for DbConnection {
+    fn import_csv(&mut self, csv_name: &str, runtime: &Runtime) {
```

Comment:
I think this _does_ probably require the Big Match Statement — the alternative would be a Struct per Dialect, which I think would be too much structure given we already have a Struct per Protocol.

---

I think it's not worth changing (even if my thought is correct, which may not be), but one question — could this be part of `IntegrationTest` trait, and have the methods receive a `protocol`. Generally implementing on the smallest unit keeps it simple. `modify_sql` doesn't even take a `protocol` arg.

Reply:
That makes sense, thanks.
I updated that.